src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -64,6 +91,59 @@ public void refreshList(String filter) {
                 .filter(pi -> pi.filter(query))
                 .sorted(Comparator.comparing(pi -> pi.username))
                 .forEach(playersModel::addElement);
+        renderer.setAddedPlayerCount(playersModel.getSize());
+        for (PlayerInfo p : nearbyPlayers) {
+            playersModel.addElement(p);
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player
+                && playersModel != null
+                && main.config.PLAYER_INFOS != null
+                && !main.config.PLAYER_INFOS.containsKey(event.getEntity().getId())) {
+
+            SwingUtilities.invokeLater(() -> {
+                int idx = indexOfPlayersModel(event.getEntity().getId());
+                if (idx < 0) {
+                    PlayerInfo playerInfo = new PlayerInfo((Player) event.getEntity());
+                    nearbyPlayers.add(playerInfo);
+                    playersModel.addElement(playerInfo);
+                } else {
+                    PlayerInfo info = playersModel.get(idx);
+                    nearbyPlayers.add(info);
```

Comment:
```suggestion
                    nearbyPlayers.add(playersModel.get(idx));
```
Slight cleanup, these can be merged